### PR TITLE
6 [Code][AI] Base GOAP System Implementation

### PR DESCRIPTION
**Pull request by Cappic-01**
Please just review the base GOAP files, not the Chase action nor the Teacher agent

### CODE REVIEW

File: Project Bunny/Assets/Scripts/AI/Actions/Chase.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            navMeshAgent.SetDestination(_kid.position);
+        }
+    
+        public override bool PrePerform()
+        {
+            var kids = Object.FindObjectsOfType<SimpleCharacterController>();
+            _kid = FindClosest(kids, navMeshAgent).transform;
+
+            if (!_kid) return false;
```
Comment (George540) [inline Project Bunny/Assets/Scripts/AI/Actions/Chase.cs:25]: Put it in brackets like normal. We'll try to avoid dangling if statements.

Comment (Cappic-01) [reply Project Bunny/Assets/Scripts/AI/Actions/Chase.cs:25]: I often put 'barrier' if statements like this for linearity and it's also how IDEs put it when you invert if statements. It's really just for if statements that only return specifically. But I don't mind putting them into brackets if that's how we prefer going with.

Comment (George540) [reply Project Bunny/Assets/Scripts/AI/Actions/Chase.cs:25]: Nah it's good, if it's for barriers, it should be fine. I think Rider recommends these pretty often too, right

**Review by George540 — APPROVED** [review]

**Review by KristianKurk — APPROVED** [review]
very nice

**Pull request merged**